### ISSUE TITLE
[BUG] Attempt to subtract with overflow

### ISSUE THREAD

**Issue by teoxoy** [dignifiedquire/async-tar#5]
I am using this code snippet to extract some files from an archived stream.

```rust
let mut ar = async_tar::Archive::new(stream_reader);
let mut entries = ar.entries()?;
use futures::stream::StreamExt;
while let Some(Ok(mut file)) = entries.next().await {
    if file
        .path()?
        .to_str()
        .map(|path| path.starts_with("some_path/"))
        .unwrap_or(false)
    {
        file.unpack_in(out_dir).await?;
    }
}
```

I am getting this error `thread 'main' panicked at 'attempt to subtract with overflow'` here:
https://github.com/dignifiedquire/async-tar/blob/0739f53aaf805f493d3952adb7d1456b5ac6715e/src/archive.rs#L361

Relevant stacktrace snippet:

```
13: core::panicking::panic
            at src/libcore/panicking.rs:52
14: async_tar::archive::poll_next_raw
            at /home/my_user/.cargo/registry/src/github.com-1ecc6299db9ec823/async-tar-0.1.1/src/archive.rs:356
15: <async_tar::archive::Entries<R> as futures_core::stream::Stream>::poll_next
            at /home/my_user/.cargo/registry/src/github.com-1ecc6299db9ec823/async-tar-0.1.1/src/archive.rs:271
16: futures_util::stream::stream::StreamExt::poll_next_unpin
            at /home/my_user/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.5/src/stream/stream/mod.rs:1323
17: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
            at /home/my_user/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.5/src/stream/stream/next.rs:35
18: std::future::poll_with_tls_context
            at /rustc/8d69840ab92ea7f4d323420088dd8c9775f180cd/src/libstd/future.rs:102
19: backend_app::download::{{closure}}
            at src/main.rs:513
```

I can see that some files are being extracted successfully but somewhere around the end of the archive, my program crashes with that error and not all files have been extracted.

**Comment by dignifiedquire** [comment]
Any chance you could provide me with an example file that triggers this?

**Comment by teoxoy** [comment]
You can try with this [archive](https://www.factorio.com/get-download/0.18.34/demo/linux64)
[Here](https://github.com/Teoxoy/factorio-blueprint-editor/blob/84e10ae921591373f782e7027fb8193a94ffc2f7/packages/backend/src/setup.rs#L120) is also the code snippet I'm using to download, decompress and untar (including the commented `async-tar` code at the end)

**Comment by Teoxoy** [comment]
You can find the code snippet I'm using to download, decompress and untar [here](https://github.com/Teoxoy/factorio-blueprint-editor/blob/84e10ae921591373f782e7027fb8193a94ffc2f7/packages/backend/src/setup.rs#L120) (including the commented `async-tar` code at the end).

**Comment by dignifiedquire** [comment]
Thank you for that, I managed to reproduce and fix finally.

**Comment by teoxoy** [comment]
Thanks for fixing it, works like a charm now.
Do you have an ETA for the next release?

**Comment by dignifiedquire** [comment]
just published `0.1.2`
